include/nana/filesystem/filesystem.hpp

```diff
@@ -30,6 +30,9 @@
 #ifndef NANA_FILESYSTEM_HPP
 #define NANA_FILESYSTEM_HPP
 
+//#undef  NANA_USING_NANA_FILESYSTEM 
+#if NANA_USING_NANA_FILESYSTEM 
```

Comment:
Hi, Jinhao..
I suspect.. not completely sure, that the linkers have some problems, maybe bugs, that when one  compiler-unit (a cpp file with all the includes) compile with one set of functions in one namespace, and other compiler-unit, have other implementation of the same functions in another namespace it have many obscure error. 
So I chose only one implementation for all the build process. It will prefer std, them boost, and then nana. 
If you need test nana, you define `NANA_FILESYSTEM_FORCE ` in the IDE, _cmake_, _make_ or in `config.h` and it will be used... well that is the idea.
If you don't care what of the three implementation to use, you can include 

`#include <nana/filesystem/filesystem_ext.hpp>  `

that is my recommendation, because it include some functions that you use, and which are not `std `now. The non `std `features are in `nana::filesystem_ext`. all the std features are in `std::experimental::filesystem`, independently of the implementation you are using.